src/app/clusters/general-diagnostics-server/general-diagnostics-server.cpp

```diff
@@ -290,6 +323,51 @@ GeneralDiagnosticsDelegate gDiagnosticDelegate;
 
 } // anonymous namespace
 
+bool emberAfGeneralDiagnosticsClusterTestEventTriggerCallback(CommandHandler * commandObj, const ConcreteCommandPath & commandPath,
+                                                              const Commands::TestEventTrigger::DecodableType & commandData)
+{
+
+    if (commandData.enableKey.size() != TestEventTriggerDelegate::kExpectedEnableKeyLength)
+    {
+        commandObj->AddStatus(commandPath, Status::ConstraintError);
+        return true;
+    }
+
+    if (IsByteSpanAllZeros(commandData.enableKey))
+    {
+        commandObj->AddStatus(commandPath, Status::ConstraintError);
+        return true;
+    }
+
+    auto * testEventTrigger = Server::GetInstance().GetTestEventTriggerDelegate();
+    if (testEventTrigger == nullptr)
+    {
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
+        return true;
+    }
+
+    if (!testEventTrigger->DoesEnableKeyMatch(commandData.enableKey))
+    {
+        commandObj->AddStatus(commandPath, Status::UnsupportedAccess);
+        return true;
+    }
+
+    Status returnStatus                 = Status::Failure;
+    CHIP_ERROR handleEventTriggerResult = testEventTrigger->HandleEventTrigger(commandData.eventTrigger);
+
+    if (handleEventTriggerResult == CHIP_NO_ERROR)
+    {
+        returnStatus = Status::Success;
+    }
+    else if (handleEventTriggerResult == CHIP_ERROR_INVALID_ARGUMENT)
```

Comment:
Suggest having an else that sets to `Failure` rather than relying on falltrhough of default `returnStatus` value